### ISSUE TITLE
Make a scilla transaction without confirmation

### ISSUE THREAD

**Issue by bogdan** [Zilliqa/Zilliqa-JavaScript-Library#162]
### **Is your feature request related to a problem? Please describe.**

In the web application development, it is a bad idea to allocate a zilliqa transaction promise because it is not resolved soon after the request is finished. It is preferred to make a periodic requests from a frontend to check status of the transaction either via backend or directly to the blockchain. 

Additionally, the application may restarted and it needs to renew the transaction tracking, while zilliqa js doesn't allow to obtain transaction id before it is confirmed

### **Describe the solution you'd like**

Allow `Contract#deploy` and `Contract#call` to skip awaiting of transactions confirmation when `attempts` is set to `0` or with an additional parameter.

Also it would good to have the follow code fragments as methods to allow to put some code between transaction creation and confirmation:

`Transaction#create`:

``` js
    const response = await this.provider.send<DeploySuccess, DeployError>(
      RPCMethod.CreateTransaction,
      { ...tx.txParams, priority: tx.toDS },
    );
    if (response.error)
       this.setStatus(TxStatus.Rejected)
     else {
        this.id = response.result.TranID
     } 
    
```

`Transation#check`:

``` js
      try {
        if (await this.trackTx(txHash)) {
          return this;
        }
      } catch (err) {
        this.status = TxStatus.Rejected;
        throw err;
      }
```

**Comment by edisonljh** [comment]
@neeboo  I recall that you were working on a feature similar to this awhile back. How's the progress?

**Comment by neeboo** [comment]
> @neeboo I recall that you were working on a feature similar to this awhile back. How's the progress?

Working slowly, a bit hard when doing the type checking and annotations.

It takes time

**Comment by neeboo** [comment]
> @neeboo I recall that you were working on a feature similar to this awhile back. How's the progress?

yep, it's actually a observed event emitter to transaction class.

use like this:

```javascript

const tx= new Transaction(...);

tx.observed().on('track',(trackInfo)=>{
   console.log(trackInfo);
}).on('receipt',(receiptObject)=>{
  console.log(receiptObject);
}).on('error',(error)=>{
  console.log(error)
});

await zilliqa.createTransaction(tx,...);


```

**Comment by bogdan** [comment]
@neeboo where is this API implemented?

**Comment by neeboo** [comment]
> @neeboo where is this API implemented?

It's inside : https://github.com/Zilliqa/Zilliqa-JavaScript-Library/blob/dev/packages/zilliqa-js-account/src/transaction.ts

in   https://github.com/Zilliqa/Zilliqa-JavaScript-Library/blob/e1b5406667323944c20b0fe5bb51e61c71aca2aa/packages/zilliqa-js-account/src/transaction.ts#L219, you can see we return a `EventEmitter`

in that case, when ever the transaction is sent, the `EventEmitter` will emit related event.

the event name is defined in https://github.com/Zilliqa/Zilliqa-JavaScript-Library/blob/e1b5406667323944c20b0fe5bb51e61c71aca2aa/packages/zilliqa-js-account/src/types.ts#L58 , and as below:

```typescript

export const enum TxEventName {
  Error = 'error',
  Receipt = 'receipt',
  Track = 'track',
}

```

**Comment by edisonljh** [comment]
Resolved with #182  

Closing.
